packages/core/src/client.ts

```diff
@@ -3820,11 +3820,21 @@ export class MedplumClient extends TypedEventTarget<MedplumClientEventMap> {
   private setupStorageListener(): void {
     try {
       window.addEventListener('storage', (e: StorageEvent) => {
-        if (e.key === null || e.key === 'activeLogin') {
-          // Storage events fire when different tabs make changes.
-          // On storage clear (key === null) or activeLogin change (key === 'activeLogin')
-          // Refresh the page to ensure the active login is up to date.
+        // Storage events fire when different tabs make changes.
+        // On storage clear (key === null) or profile change (key === 'activeLogin', and profile in 'activeLogin' is different)
+        // Refresh the page to ensure the active login is up to date.
+        if (e.key === null) {
           window.location.reload();
+        } else if (e.key === 'activeLogin') {
+          try {
```

Comment:
Do we need this `try`/`catch`? should only happen on malformed JSON (which generally shouldn't happen).  I think (?) that will result in same behavior, an unhandled error in the console.

Reply:
Good point, I removed it 👍